modules/transformer-batch/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/transformer/batch/SparkSchema.scala

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2012-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and
+ * limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.transformer.batch
+
+import cats.Monad
+import cats.data.EitherT
+import cats.implicits._
+import cats.effect.Clock
+
+import com.snowplowanalytics.iglu.schemaddl.parquet.{Field, Type}
+import com.snowplowanalytics.iglu.client.Resolver
+import com.snowplowanalytics.iglu.client.resolver.registries.RegistryLookup
+import com.snowplowanalytics.snowplow.analytics.scalasdk.SnowplowEvent
+import com.snowplowanalytics.snowplow.rdbloader.common.LoaderMessage
+import com.snowplowanalytics.snowplow.rdbloader.common.transformation.WideField
+
+import org.apache.spark.sql.types._
+
+object SparkSchema {
+
+  val CustomDecimal = DecimalType(18, 2)
+
+  def build[F[_]: Clock: Monad: RegistryLookup](resolver: Resolver[F], types: List[LoaderMessage.TypesInfo.WideRow.Type]): F[Either[String, StructType]] = {
+    val latestByModel = LoaderMessage.TypesInfo.WideRow.latestByModel(types)
+    latestByModel.sorted
+      .traverse(forEntity(resolver, _))
+      .map(forEntities => StructType(Atomic ::: forEntities))
+      .value
+  }
+
+  def forEntity[F[_]: Clock: Monad: RegistryLookup](resolver: Resolver[F], entity: LoaderMessage.TypesInfo.WideRow.Type): EitherT[F, String, StructField] =
+    WideField.getSchema(resolver, entity.schemaKey).map { schema =>
+      val name = SnowplowEvent.transformSchema(entity.snowplowEntity.toSdkProperty, entity.schemaKey)
+      entity.snowplowEntity match {
+        case LoaderMessage.SnowplowEntity.SelfDescribingEvent =>
+          // TODO: Should 'Field.normalized' be called here ?
```

Comment:
I'm fairly sure yes it does need to be called.  Would be great to add a field to [these test schemas](https://github.com/snowplow/snowplow-rdb-loader/tree/054e98e164987bfa249c0ea4e37919125a0acc4a/modules/transformer-batch/src/test/resources/widerow/parquet/schemas/com.snowplowanalytics.snowplow/parquet_test_a/jsonschema) with a weird name, so we can assert the parquet column is written properly.